### PR TITLE
Move mapping out of module executor

### LINKED ISSUE #199: Refactor ModuleRegistry
## The Problem
The init method of the Module registry is currently a huge mess of functions that map between Library classes and our data classes

## Possible Solution
Those mapping functions should be extracted in seperate and tidy functions and should be moved to a new Kotlin file, probably in a newly created `jira` package shared between this new file and the currently named `Jira.kt`

### PR DESCRIPTION
**Pull request by urielsalis**
## Purpose
Move mapping out of module executor. Fixes #199
## Approach
Centralize the domain of modules instead of duplicating
Reusing the same functions
Move the functions to a jira package

## Future work
- Use Instant everywhere #206
- Add tests to all mapping functions #207

#### Checklist
- [ ] Included tests
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

File: src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package io.github.mojira.arisa.domain
+
+import arrow.core.Either
+import java.util.Date
+
+data class Attachment(
+    val name: String,
+    val created: Date,
```
Comment (NeunEinser) [inline src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt:8]: That might have been inconsistent already, but we probably should change all of these to Instant for consistency.

File: src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package io.github.mojira.arisa.domain
+
+data class ChangeLogItem(
+    val created: Long,
```
Comment (NeunEinser) [inline src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt:4]: Probably for Long too, as Instant provides an easy function to obtain millis anyways. It just helps with what you should expect when seeing created / updated.

File: src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt [hunk 3]
```diff
@@ -203,22 +208,24 @@ class ModuleRegistry(jiraClient: JiraClient, private val config: Config) {
...
-             AbstractTransferFieldModule.Request(
-                 issue.key,
-                 links,
-                 links
-             )
-         }*/
+            AbstractTransferFieldModule.Request(
+                issue.key,
+                links,
+                links
```
Comment (NeunEinser) [inline src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt]: If you have a better idea for that module without reusing too much code between this and TransferVersions, feel free btw 😉

File: src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt (hunk 1 above)
Comment (urielsalis) [reply src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt:8]: Indeed

File: src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt (hunk 2 above)
Comment (urielsalis) [reply src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt]: I would argue that we should use Instant everywhere

**Comment by NeunEinser** [comment]
Apart from tests everything compilles now 😅

File: src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt (hunk 2 above)
Comment (NeunEinser) [reply src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt]: I actually already changed it to Instant when fixing build errors

File: src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt (hunk 1 above)
Comment (urielsalis) [reply src/main/kotlin/io/github/mojira/arisa/domain/Attachment.kt:8]: https://github.com/mojira/arisa-kt/issues/206

File: src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt (hunk 2 above)
Comment (urielsalis) [reply src/main/kotlin/io/github/mojira/arisa/domain/ChangeLogItem.kt]: https://github.com/mojira/arisa-kt/issues/206

File: src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt [hunk 4]
```diff
@@ -415,7 +258,11 @@ class ModuleRegistry(jiraClient: JiraClient, private val config: Config) {
                             defaultMessage
                         ) else defaultMessage
 
-                    addRestrictedComment(issue, text, "helper")
+                    addRestrictedComment(
+                        issue,
+                        text,
+                        "helper"
+                    )
```
Comment (SPGoding) [inline src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt:265]: Imo we can just leave this as-is. I plan to refactor the adding comment part when resolving #139; this whole part could be replaced by a single `partially` function reference at that time.

File: src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+private fun JiraIssue.getFieldAsString(field: String) = this.getField(field) as? String?
+
+private fun JiraIssue.getCustomField(customField: String): String? =
+    ((getField(customField)) as? JSONObject)?.get("value") as? String?
+
+fun JiraIssue.getEnvironment() = getFieldAsString("environment")
+
+fun JiraIssue.getCHK(config: Config) = getFieldAsString(config[Arisa.CustomFields.chkField])
+fun JiraIssue.getConfirmation(config: Config) = getCustomField(config[Arisa.CustomFields.confirmationField])
```
Comment (SPGoding) [inline src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt:40]: There are several duplicated codes of converting `null` or empty confirmation status to `Unconfirmed` (like the `String#getOrDefault` extension function in `RevokeConfirmationModule`).

Will it make sense to move that conversion here?

File: src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt [hunk 6]
```diff
@@ -22,7 +23,16 @@ class RemoveNonStaffMeqsModuleTest : StringSpec({
...
+        val comment = Comment(
+            "I like QC.",
+            "",
+            { emptyList() },
+            Instant.now(),
+            Instant.now(),
+            null,
+            null,
+            { Unit.right() },
+            { Unit.right() })
```
Comment (SPGoding) [inline src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt:35]: (Trivial) (Subjective) (WAI) The last parentheses drives me crazy lol

File: src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt [hunk 7]
```diff
@@ -333,7 +333,8 @@ class CrashModuleTest : StringSpec({
             crashReader
         )
 
-        val attachment = Attachment("crash.txt", Calendar.getInstance().time) { EXAMPLE_CRASH.toByteArray() }
+        val attachment =
+            Attachment("crash.txt", Calendar.getInstance().time, { Unit.right() }, { EXAMPLE_CRASH.toByteArray() })
```
Comment (violine1101) [inline src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt:337]: (Minor) (Trivial) I don't like that line break here

File: src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt (hunk 6 above)
Comment (violine1101) [reply src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt:35]: Don't we have a linter for this sort of stuff? Is it not working properly, or does it not have a rule for this sort of stuff?

File: src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt (hunk 7 above)
Comment (NeunEinser) [reply src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt:337]: I'd suspect this is an automatic thing by IntelliJ. I do prefer to have parameters on seperate lines as well tho.

File: src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt (hunk 6 above)
Comment (NeunEinser) [reply src/test/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffMeqsModuleTest.kt:35]: No rule, apparently

File: src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt (hunk 5 above)
Comment (NeunEinser) [reply src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt:40]: (Example: A module that sets to Unconfirmed when it's set to null or empty)

**Comment by NeunEinser** [comment]
I think we should merge this tomorrow (so that people are actually awake to notice any issues or things that might have broken with this)

File: src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt (hunk 7 above)
Comment (urielsalis) [reply src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt:337]: It was automatically added by intellij as the line is too long 😅

**Pull request merged**